src/paper/tasks.py

```diff
@@ -750,98 +753,214 @@ def pull_biorxiv_papers():
+            entry_id = result.entry_id.split("http://arxiv.org/abs/")[1]
+            pure_doi = "arXiv.{}".format(entry_id.split("v")[0])
+            title = result.title
+            abstract = result.summary
+            publication_date = None
+            raw_authors = []
+            hubs = []
+            publication_date = result.published
+            for arxiv_cat in result.categories:
+                cur_cat = get_category_name(arxiv_cat)
+                hubs.append(cur_cat)
+
+            for author in result.authors:
                 try:
-                    paper = Paper(**data)
-                    paper.full_clean()
-                    paper.save()
-
-                    concept_names = [
-                        concept.get("display_name", "other")
-                        for concept in concepts
-                        if concept.get("level", 0) == 0
-                    ]
-                    potential_hubs = []
-                    for concept_name in concept_names:
-                        potential_hub = Hub.objects.filter(name__icontains=concept_name)
-                        if potential_hub.exists():
-                            potential_hub = potential_hub.first()
-                            potential_hubs.append(potential_hub)
-                            hub_ids.add(potential_hub.id)
-                    paper.hubs.add(*potential_hubs)
-                    download_pdf.apply_async((paper.id,), priority=4, countdown=4)
+                    author_split = author.name.split(" ")
+                    author_name_length = len(author_split)
+                    if author_name_length == 1:
+                        first_name = author_split[0]
+                        last_name = ""
+                    elif author_name_length == 2:
+                        first_name = author_split[0]
+                        last_name = author_split[1]
+                    elif author_name_length > 2:
+                        last_name = author_split[-1]
+                        first_name = ""
+                        for name_index, name_part in enumerate(author_split):
+                            if name_index != author_name_length - 1:
+                                first_name += name_part + " "
+                        first_name = first_name.strip()
+                    raw_authors.append(
+                        {"first_name": first_name, "last_name": last_name}
+                    )
                 except Exception as e:
+                    print(e)
                     sentry.log_error(e)
-        arxiv_works = open_alex.get_data_from_source(arxiv_id, yesterday, page=i + 1)
+
+            data = {
+                "doi": pure_doi,
+                "url": result.entry_id,
+                "raw_authors": format_raw_authors(raw_authors),
+                "title": title,
+                "paper_title": title,
+                "paper_publish_date": publication_date,
+                "is_open_access": True,
```

Comment:
Hard coding open access feels iffy without a license check.

Reply:
There's no license info unfortunately in the API (or at least I couldn't find it). 

I think is_open_access is always true here -- it's just some articles may have a different open_access status. But their api for some reason doesn't expose this.